modAion/src/org/aion/zero/db/AionContractDetailsImpl.java

```diff
@@ -72,18 +70,32 @@ public void put(ByteArrayWrapper key, ByteArrayWrapper value) {
         // We strip leading zeros of a DataWord but not a DoubleDataWord so that when we call get
         // we can differentiate between the two.
 
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(value);
+
         if (value.isZero()) {
-            storageTrie.delete(key.getData());
-        } else {
-            // TODO: VM must handle padding
-            boolean isDouble = value.getData().length == DoubleDataWord.BYTES;
-            byte[] data =
-                    (isDouble)
-                            ? RLP.encodeElement(value.getData())
-                            : RLP.encodeElement(value.getNoLeadZeroesData());
-            storageTrie.update(key.getData(), data);
+            // TODO: remove when integrating the AVM
+            // used to ensure FVM correctness
+            throw new IllegalArgumentException(
+                    "Put with zero values is not allowed for the FVM. Explicit call to delete is necessary.");
         }
 
+        // TODO: VM must handle padding
+        boolean isDouble = value.getData().length == DoubleDataWord.BYTES;
+        byte[] data =
+                (isDouble)
+                        ? RLP.encodeElement(value.getData())
+                        : RLP.encodeElement(value.getNoLeadZeroesData());
+        storageTrie.update(key.getData(), data);
+
+        this.setDirty(true);
+        this.rlpEncoded = null;
+    }
+
+    @Override
+    public void delete(ByteArrayWrapper key) {
+        storageTrie.delete(key.getData());
```

Comment:
should we check the key?
Objects.requireNonNull(key);

Reply:
Yes, we should. I will make the update.

Reply:
Okay, make sure the other operation methods get checked.